homeassistant/components/climate/honeywell.py

```diff
@@ -210,6 +210,13 @@ def update(self):
             self._name = data['name']
             self._is_dhw = False
 
+        # The underlying library doesn't expose the thermostat's mode
+        # but we can pull it out of the big dictionary of information.
+        device = self.client.named_devices[self.name]
```

Comment:
I'm thinking about the case when the device is a "DOMESTIC_HOT_WATER" thermostat. Since we set `self._name` to 'Hot Water' in that case, I don't think we can use the name to get the device in that case. We could use the id instead, which will work for all cases.
```py
device = self.client.devices[self._id]
```
But will a "DOMESTIC_HOT_WATER" device have `device['thermostat']['changeableValues']['mode']` accessible?